Design/SANS/BeamCentreFinderDesign.md

```diff
@@ -0,0 +1,271 @@
+# Design Document for Beam Centre Finder
+
+### Current issue
+In order to calibrate the Beam Centre position with respect to the detectors, an accurate estimation estimation of the position needs to be performed. This functionality is provided by a Beam Centre Finder (*BCF*). 
+
+An implementation for this BCF already exists for ISIS SANS. This implementation is heavily tied up with the ISIS instruments and the specific way the data reduction is realized for ISIS SANS. ANSTO requires the same functionality as the current *BCF* provides. Unfortunately, it is not possible to directly use the *BCF*'s functionality with the ANSTO specific implementation of data reduction.
+
+The different approaches of ISIS and ANSTO are currently not compatible. They can be summarized as:
+
+* ISIS makes use of a so-called ReductionSingleton which has a series of registered reduction steps. During the reduction, the ReductionSingleton() carries state (mainly via the custom ISIS instrument classes). Intermediate results are stored in the Analysis Data Service (ADS).
+
+* ANSTO's reduction implementation is currently still work in progress, but will eventually make use of so-called Workflow-Algorithms which do not populate the ADS or carry state information. This will be a cleaner, more maintainable implementation which is easier to test and verify. It is desirable that in the long run ISIS SANS switches to such an implementation.
+
+This document outlines the required changes to make the *BCF* useable in a more general, faciltity- and instrument-agnostic way.
+
+The benefits of introducing a more general BCF are:
+
+* Avoiding a facility-based implementation and having this functionality in a dedicated location will make the BCF much more maintainable and it will be easier to introduce changes which affect all facilities.
+* A *BCF* which does not depend on facility- and instrument-specifics will make this functionality reusable for new instruments and facilities which might join Mantid in the future.
+* Removing the dependence of the ReductionSingleton will allow for easier unit testing of the BCF.We currently have one system test and zero unit tests for the *BCF* functionality. 
+
+
+The static details of the current implementation (ie a class diagram) can be seen here:
+
+![alt text](beam_centre_finder_classes.png)
+
+
+The workflow of the current implementation is illustrated below. It is separated into two parts, one which is repsonsible for stepping through the parameter space
+![alt text](FindBeamCentre_py.png)
+
+and one which illustrates how the residuals -- a measure of our search quality -- are calculated. This is achieved by the *SeekCentre* operation:
```

Comment:
I don't think that any mention of `SeekCenter` is made before this. We do however have a `CenterFinder` in the diagram.